geth/rpc/route.go

```diff
@@ -0,0 +1,72 @@
+package rpc
+
+// router implements logic for routing
+// JSON-RPC requests either to Upstream or
+// Local node.
+type router struct {
+	methods         map[string]bool
+	upstreamEnabled bool
+}
+
+// newRouter inits new router.
+func newRouter(upstreamEnabled bool) *router {
+	r := &router{
+		methods:         make(map[string]bool),
+		upstreamEnabled: upstreamEnabled,
+	}
+
+	for _, m := range localMethods {
+		r.methods[m] = true
+	}
+
+	return r
+}
+
+// routeLocally returns true if given method should be routed to
+// the local node
+func (r *router) routeLocally(method string) bool {
+	// if upstream is disabled, always route to
+	// the local node
+	if !r.upstreamEnabled {
+		return true
+	}
+
+	// else check route using the methods list
+	return r.methods[method]
+}
+
+// localMethods contains methods that should be routed to
+// the local node; the rest is considered to be routed to
+// the upstream node.
+var localMethods = [...]string{
```

Comment:
Please, add a TODO that we want to rename it to `remoteMethods` for some safety. If there's a new method in geth, we'll route it to the local node instead of the remote, which adds some reliability.

Reply:
Didn't know that, thanks!